packages/samples/headless-ssr/src/components/hydration-metadata.tsx

```diff
@@ -0,0 +1,33 @@
+import {FunctionComponent} from 'react';
+import {SearchCSRState, SearchSSRState} from '../common/engine';
+
+export interface HydrationMetadataProps {
+  initialState: SearchSSRState;
+  hydrationResult?: SearchCSRState;
+}
+
+export const HydrationMetadata: FunctionComponent<HydrationMetadataProps> = ({
+  initialState,
+  hydrationResult,
+}) => (
+  <>
+    <div>
+      Hydrated:{' '}
+      <span id="hydrated-indicator">{hydrationResult ? 'yes' : 'no'}</span>
+    </div>
+    <span id="hydrated-msg">
+      Rendered engine with{' '}
+      {
+        (hydrationResult ?? initialState).controllers.resultList.state.results
```

Comment:
Logically wouldn't the num of results make more sense to be part of result list cmp rather than hydration metadata cmp?